jib-core/src/main/java/com/google/cloud/tools/jib/api/JavaContainerBuilder.java

```diff
@@ -0,0 +1,291 @@
+        layerConfigurationsBuilder.addDirectoryContents(
+            LayerType.DEPENDENCIES,
+            file,
+            path -> !path.getFileName().toString().contains("SNAPSHOT"),
+            DEPENDENCIES_PATH);
+        layerConfigurationsBuilder.addDirectoryContents(
+            LayerType.SNAPSHOT_DEPENDENCIES,
+            file,
+            path -> path.getFileName().toString().contains("SNAPSHOT"),
+            DEPENDENCIES_PATH);
+      } else {
+        layerConfigurationsBuilder.addFile(
+            file.getFileName().toString().contains("SNAPSHOT")
+                ? LayerType.SNAPSHOT_DEPENDENCIES
+                : LayerType.DEPENDENCIES,
+            file,
+            DEPENDENCIES_PATH.resolve(file.getFileName()));
+      }
+    }
+    if (!classpath.contains(DEPENDENCIES_PATH.resolve("*").toString())) {
+      classpath.add(DEPENDENCIES_PATH.resolve("*").toString());
+    }
+    return this;
+  }
+
+  /**
+   * Adds dependencies to /app/libs on the image.
+   *
+   * @param dependencyFiles the list of dependencies to add to the image
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addDependencies(Path... dependencyFiles) throws IOException {
+    return addDependencies(Arrays.asList(dependencyFiles));
+  }
+
+  /**
+   * Adds resources to /app/resources on the image.
+   *
+   * @param resourceFilesDirectory the directory containing the project's resources
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addResources(Path resourceFilesDirectory) throws IOException {
+    if (!Files.exists(resourceFilesDirectory) || !Files.isDirectory(resourceFilesDirectory)) {
+      throw new IOException(
+          "Adding resources failed: '" + resourceFilesDirectory + "' is not a directory");
+    }
+    layerConfigurationsBuilder.addDirectoryContents(
+        LayerType.RESOURCES, resourceFilesDirectory, path -> true, RESOURCES_PATH);
+    if (!classpath.contains(RESOURCES_PATH.toString())) {
+      classpath.add(RESOURCES_PATH.toString());
+    }
+    return this;
+  }
+
+  /**
+   * Adds classes to /app/classes on the image.
+   *
+   * @param classFilesDirectory the directory containing the class files
+   * @return this
+   * @throws IOException if adding the layer fails
+   */
+  public JavaContainerBuilder addClasses(Path classFilesDirectory) throws IOException {
+    if (!Files.exists(classFilesDirectory) || !Files.isDirectory(classFilesDirectory)) {
+      throw new IOException(
+          "Adding classes failed: '" + classFilesDirectory + "' is not a directory");
+    }
+    layerConfigurationsBuilder.addDirectoryContents(
+        LayerType.CLASSES, classFilesDirectory, path -> true, CLASSES_PATH);
+    if (!classpath.contains(CLASSES_PATH.toString())) {
+      classpath.add(CLASSES_PATH.toString());
+    }
+    return this;
+  }
+
+  /**
+   * Adds additional files to the image's classpath.
```

Comment:
I think we can remove `image's` since the classpath is for the java invocation and not the image itself.